### PR TITLE
[SPARK-27103][SQL][MINOR] List SparkSql reserved keywords in alphabet order

### PR DESCRIPTION
**Pull request by SongYadong**
## What changes were proposed in this pull request?

This PR tries to correct spark-sql reserved keywords' position in list if they are not in alphabetical order. 
In test suite some repeated words are removed. Also some comments are added for remind.

## How was this patch tested?

Existing unit tests.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala [hunk 1]
```diff
@@ -23,39 +23,39 @@ class TableIdentifierParserSuite extends SparkFunSuite {
...
+    "purge", "query", "range", "read", "readonly", "reads", "rebuild", "recordreader",
+    "recordwriter", "regexp", "reload", "rename", "repair", "replace", "replication", "restrict",
+    "revoke", "rewrite", "rlike", "role", "roles", "rollup", "row", "rows", "schemas", "second",
+    "serde", "serdeproperties", "server", "set", "sets", "shared", "show", "show_database",
+    "skewed", "smallint", "snapshot", "sort", "sorted", "ssl", "statistics", "stored",
+    "streamtable", "string", "struct", "table", "tables", "tblproperties", "temporary",
+    "terminated", "timestamp", "tinyint", "to", "touch", "trailing", "transaction", "transactions",
+    "trigger", "true", "truncate", "unarchive", "undo", "uniontype", "unlock", "unset", "unsigned",
+    "update", "uri", "use", "user", "utc", "utctimestamp", "values", "view", "while", "with",
+    "work", "write", "year")
```
Comment (SongYadong) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala]: put them in alphabet order and remove some repeated words, including:
`"int", "smallint", "double", "float", "timestamp", "binary", "boolean", "both", "current_date", "current_timestamp", "date"`

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala [hunk 2]
```diff
@@ -23,39 +23,39 @@ class TableIdentifierParserSuite extends SparkFunSuite {
...
 
-  val hiveStrictNonReservedKeyword = Seq("anti", "full", "inner", "left", "semi", "right",
-    "natural", "union", "intersect", "except", "database", "on", "join", "cross", "select", "from",
-    "where", "having", "from", "to", "table", "with", "not")
+  val hiveStrictNonReservedKeyword = Seq("anti", "cross", "database", "except", "from", "full",
+    "having", "inner", "intersect", "join", "left", "natural", "not", "on", "right", "select",
+    "semi", "table", "to", "union", "where", "with")
```
Comment (SongYadong) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala]: put them in alphabet order and remove one repeated word: `"from"`

**Review by srowen — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
cc @maropu

**Comment by maropu** [comment]
The change itself looks ok to me. But, I feel its a bit cumbersome to always keep this alphabetical order and duplicated entries easily happen by the current hard-to-check format (multiple keywords in a line).

For example, postgresql place a single keyword in a line: https://github.com/postgres/postgres/blob/277cb789836b5ddf81aabb80c2058268c70e2f36/src/backend/parser/gram.y#L14975 (this format makes the number of lines brow up though...)

**Comment by dongjoon-hyun** [comment]
Is it a good time to transform into PostgrSQL style? For me, +1 for that kind of transition for the future.

cc @hvanhovell , too.

**Comment by SongYadong** [comment]
Indeed it's clearer in PostgreSQL style. But that makes files too long. I'm not sure how to balance.

**Comment by srowen** [comment]
I don't mind letting this file grow long if it helps readability by adopting PostgreSQL style.
I don't think it has to block the cleanup here, in any event.

**Comment by SongYadong** [comment]
I got it. Let me try now.

**Comment by SongYadong** [comment]
Updated. 
cc @srowen @dongjoon-hyun @maropu

**Comment by viirya** [comment]
Looks good. For such change, it might be worth creating a JIRA for this.

**Comment by SongYadong** [comment]
@viirya thanks. I will do that.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM, too. Merged to master.
Thank you, @SongYadong , @srowen , @maropu , @viirya .

**Comment by SongYadong** [comment]
Thanks for review.  @srowen  @dongjoon-hyun @maropu @viirya

**Pull request closed without merging**